packages/jaeger-ui/src/components/SearchTracePage/SearchForm.jsx

```diff
@@ -259,44 +259,79 @@ export function submitForm(fields, searchTraces) {
 }
 
 export class SearchFormImpl extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formData: {
+        service: this.props.initialValues?.service,
+        operation: this.props.initialValues?.operation,
+        tags: this.props.initialValues?.tags,
+        lookback: this.props.initialValues?.lookback,
+        startDate: this.props.initialValues?.startDate,
+        startDateTime: this.props.initialValues?.startDateTime,
+        endDate: this.props.initialValues?.endDate,
+        endDateTime: this.props.initialValues?.endDateTime,
+        minDuration: this.props.initialValues?.minDuration,
+        maxDuration: this.props.initialValues?.maxDuration,
+        resultsLimit: this.props.initialValues?.resultsLimit,
+      },
+    };
+  }
+
+  handleChange = fieldData => {
+    this.setState(prevState => ({
+      formData: {
+        ...prevState.formData,
+        ...fieldData,
+      },
+    }));
+    if (fieldData.service) {
+      this.props.changeServiceHandler(fieldData.service);
+      this.setState(prevState => ({
+        formData: {
+          ...prevState.formData,
+          operation: DEFAULT_OPERATION,
+        },
+      }));
+    }
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.submitFormHandler(this.state.formData);
+  };
+
   render() {
-    const {
-      handleSubmit,
-      invalid,
-      searchMaxLookback,
-      selectedLookback,
-      selectedService = '-',
-      services,
-      submitting: disabled,
-    } = this.props;
+    const { invalid, searchMaxLookback, services, submitting: disabled } = this.props;
```

Comment:
Personally I think it should always be called `submitting`, to reflect the state of the form. `disabled` is just a property of individual fields.